src/consensus/block_body_validation.py

```diff
@@ -148,12 +148,16 @@ async def validate_block_body(
     if height <= constants.INITIAL_FREEZE_PERIOD and block.transactions_generator is not None:
         return Err.INITIAL_TRANSACTION_FREEZE, None
 
+    # transactions_generator_ref_list may be None, even if transactions_generator is not None
     if height > constants.INITIAL_FREEZE_PERIOD and constants.NETWORK_TYPE == NetworkType.MAINNET:
         if block.transactions_generator is not None:
             if len(bytes(block.transactions_generator)) > constants.MAX_GENERATOR_SIZE:
                 return Err.PRE_SOFT_FORK_MAX_GENERATOR_SIZE, None
-            else:
-                return None, None
+            elif block.transactions_generator_ref_list is not None:
+                if len(bytes(block.transactions_generator_ref_list)) > constants.MAX_GENERATOR_REF_LIST_SIZE:
```

Comment:
Take out `bytes` and make the underlying structure `List[int]` so we can limit the number of referenced generators rather than the size of this structure.